src/Lean/Meta/Constructions/NoConfusionLinear.lean

```diff
@@ -68,24 +81,49 @@ def mkWithCtorName (indName : Name) : Name :=
 def mkNoConfusionTypeName (indName : Name) : Name :=
   Name.str indName "noConfusionType"
 
+private partial def getKindLevel (indTyKind : Expr) : MetaM Level := do
+  match (← whnf indTyKind) with
+  | .sort l => decLevel l
```

Comment:
Late night thought: This will probably not work with a type former like `ULift` with a level that is definitely not 0, but cannot be decreased.

So maybe I need the more general PULift that takes two sort levels after all.